src/Aspire.Hosting/Publishing/ManifestPublishingContext.cs

```diff
@@ -297,4 +268,65 @@ internal void WriteManifestMetadata(IResource resource)
 
         Writer.WriteEndObject();
     }
+
+    private void WriteContainerMounts(ContainerResource container)
+    {
+        if (container.TryGetAnnotationsOfType<ContainerMountAnnotation>(out var mounts))
+        {
+            // Write out details for bind mounts
+            var bindMounts = mounts.Where(mounts => mounts.Type == ContainerMountType.Bind).ToList();
```

Comment:
Is there a way to exclude a bind mount from the manifest? Kind of like the "RunAs" we have for other commands?

The scenario I'm thinking about is like what we have for DatabaseContainers in the samples:

https://github.com/dotnet/aspire-samples/blob/preview.4/samples/DatabaseContainers/DatabaseContainers.AppHost/Program.cs#L17-L23

I wouldn't really want these bind mounts in production, right?

Reply:
I considered proposing adding a bool to `WithBindMount` to make this easier but decided that for now folks can just add the `if` statement themselves. If we see that pattern become really popular we can consider baking a method into `Aspire.Hosting` to make it more formal.

Reply:
TBC, we could consider adding `RunWithBindMount`/`PublishWithBindMount`, or `WithBindMount(string source, string target, bool isReadOnly = false, bool includeInManifest = true)` in the future, but until then folks can just do:

```csharp
var pg = builder.AddPostgres("pg");
if (builder.ExecutionContext.IsRunning)
{
    pg.WithBindMount("./pgdata", "/var/lib/postgresql/data");
}
var db = pg.AddDatabase("db")
```